File: 이코테 복습(python, c++, java)/그리디/python/1이 될 때까지.py
n, k = map(int, input().split())
cnt = 0

while True:
    target = (n // k) * k
    cnt += n - target
    n = target
    if n < k:
        cnt += n - 1
        break
    n //= k
    cnt += 1

print(cnt)

File: 이코테 복습(python, c++, java)/그리디/c++/1이 될 때까지.cpp
#include<iostream>
using namespace std;

int n, k;
int cnt;

int main() {
	cin >> n >> k;
	while (true) {
		int target = (n / k) * k;
		cnt += n - target;
		n = target;

		if (n < k) {
			cnt += (n - 1);
			break;
		}

		n /= k;
		cnt++;
	}
	cout << cnt << endl;
}